test/ptc/unit/start_test.clj

```diff
@@ -1,46 +1,29 @@
 (ns ptc.unit.start-test
   (:require [clojure.test :refer [deftest is testing]]
-            [clojure.edn  :as edn]
-            [ptc.start    :as start]
-            [ptc.util.jms :as jms])
-  (:import [org.apache.activemq ActiveMQSslConnectionFactory]))
-
-;; Local testing for ActiveMQ
-;; https://activemq.apache.org/how-do-i-embed-a-broker-inside-a-connection
-;;
-(defn with-test-jms-connection
-  "CALL with a local JMS connection for testing."
-  [call]
-  (let [url     "vm://localhost?broker.persistent=false"
-        factory (new ActiveMQSslConnectionFactory url)
-        queue   "test.queue"]
-    (with-open [connection (.createQueueConnection factory)]
-      (call connection queue))))
-
-(def message
-  "Example JMS message for testing."
-  (edn/read-string (slurp "test/data/good-jms.edn")))
+            [ptc.start :as start]
+            [ptc.util.jms :as jms]
+            [ptc.tools.jms :refer [with-test-queue-connection message]]))
 
 (deftest produce-consume-message
-  (let [text "consume"
-        properties (::jms/Properties (jms/encode message))]
+  (let [text       "consume"
+        properties (::jms/Properties (jms/encode @message))]
     (letfn [(produce-consume [connection queue]
               (start/produce connection queue text properties)
               (start/consume connection queue))]
-      (if-let [msg (with-test-jms-connection produce-consume)]
+      (if-let [msg (with-test-queue-connection produce-consume)]
         (testing "Message is not nil and can be properly consumed"
           (is (= text (.getText msg)))
-          (is (apply = (map ::jms/Properties [message (jms/ednify msg)]))))
+          (is (apply = (map ::jms/Properties [@message (jms/ednify msg)]))))
         (is false)))))
 
 (deftest produce-peek-message
-  (let [text "peek"
-        properties (::jms/Properties (jms/encode message))]
+  (let [text       "peek"
+        properties (::jms/Properties (jms/encode @message))]
     (letfn [(produce-peek [connection queue]
               (start/produce connection queue text properties)
               (start/peek-message connection queue))]
-      (if-let [msg (with-test-jms-connection produce-peek)]
+      (if-let [msg (with-test-queue-connection produce-peek)]
         (testing "Message is not nil and can be properly consumed"
           (is (= text (.getText msg)))
-          (is (apply = (map ::jms/Properties [message (jms/ednify msg)]))))
+          (is (apply = (map ::jms/Properties [@message (jms/ednify msg)]))))
         (is false)))))
```

Comment:
Maybe just inline the message slurp?

Reply:
I'm not sure what you mean? Can you give an example?

Reply:
`[(edn/read-string (slurp "test/data/good-jms.edn")) (jms/ednify msg)]`

… as opposed to `(defn get-good-msg …)` and then

`[(get-good-msg) (jms/ednify msg)]`

Not a big deal either way though.

Reply:
That un-namespaced `@message` looks weird to me,
but maybe I should just get over it.

Reply:
I've removed the un-namespaced use